packages/swap/src/utils/intercept.ts

```diff
@@ -0,0 +1,26 @@
+import { Request, Response, NextFunction } from 'express';
+import prisma from '../client';
+
+export const getLastChainTrackingUpdateTimestamp = async () => {
+  const latestChainTracking = await prisma.chainTracking.findFirst({
+    orderBy: {
+      updatedAt: 'desc',
+    },
+  });
+
+  if (!latestChainTracking) {
+    return undefined;
+  }
+
+  return latestChainTracking.updatedAt;
+};
+
+export const stalenessCheck = async (_: Request, res: Response, next: NextFunction) => {
```

Comment:
need to use `asyncHandler` or try/catch. if one of these promises rejects, the rejected promise isn't handled by express >:(

Reply:
uf, good point!